### ISSUE TITLE
Latest version: nil pointer on .Release.Name

### ISSUE THREAD

**Issue by samudurand** [airflow-helm/charts#116]
**What is the bug?**

When using the latest version released for Airflow 2 (v8.0.2), and using the `template` command, I obtain a nil exception on .Release.Name

```
Error: UPGRADE FAILED: template: airflow/charts/airflow/templates/worker/worker-statefulset.yaml:170:22: executing "airflow/charts/airflow/templates/worker/worker-statefulset.yaml" at <include "airflow.volumes" (dict "Values" .Values "extraPipPackages" $extraPipPackages "extraVolumes" $extraVolumes)>: error calling include: template: airflow/charts/airflow/templates/_helpers/pods.tpl:238:19: executing "airflow.volumes" at <include "airflow.fullname" .>: error calling include: template: airflow/charts/airflow/templates/_helpers/common.tpl:6:24: executing "airflow.fullname" at <.Release.Name>: nil pointer evaluating interface {}.Name
helm.go:94: [debug] template: airflow/charts/airflow/templates/worker/worker-statefulset.yaml:170:22: executing "airflow/charts/airflow/templates/worker/worker-statefulset.yaml" at <include "airflow.volumes" (dict "Values" .Values "extraPipPackages" $extraPipPackages "extraVolumes" $extraVolumes)>: error calling include: template: airflow/charts/airflow/templates/_helpers/pods.tpl:238:19: executing "airflow.volumes" at <include "airflow.fullname" .>: error calling include: template: airflow/charts/airflow/templates/_helpers/common.tpl:6:24: executing "airflow.fullname" at <.Release.Name>: nil pointer evaluating interface {}.Name
UPGRADE FAILED
main.newUpgradeCmd.func2
	/home/circleci/helm.sh/helm/cmd/helm/upgrade.go:156
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main
	/home/circleci/helm.sh/helm/cmd/helm/helm.go:93
runtime.main
	/usr/local/go/src/runtime/proc.go:203
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1373
```

The command I I use is:
```
helm3 template airflow [... options]
```
The dependency on Release.Name has been introduced in the version 8.0.0

Note:
I have also tried `helm3 template airflow airflow [... options]` (to give an explicit name)

**Comment by bensta** [comment]
I got the same issue trying to install it using the helm go client library. But the issue seems to depend on the values file as it works with some configs and doesn't with others. Currently I don't know which config causes the issue for me as I didn't have the time to debug in detail.

**Comment by samudurand** [comment]
Yes you could be right, it might be related to the extraVolumes or extraPipPackages. I am not using either.

**Comment by bensta** [comment]
Yep, that rings a bell. I do use both of them though. extraPipPackages globally (under airflow.extraPipPackages) and extraVolumes for the webserver only (under web.extraVolumes). I think it showed up when I added configs there. Could be a coincidence though.

**Comment by samudurand** [comment]
It also happens with the `upgrade` command, which is a much bigger issue...

**Comment by samudurand** [comment]
Ok so ... the problem is completely unrelated, it's because of the "knownHosts" configuration. If you use it as recommended in the docs, this breaks the build with the error mentioned above.

```
      sshKnownHosts: |-
        github.com ssh-rsa xxxxxxxxxx
```

**Comment by bensta** [comment]
did some debugging: The culprit seems to be the gitSync.sshKnownHosts config. 
It **works** when:
- dags.gitSync.sshKnownHosts is empty + my (quite extensive) additional values configuring our deployment 
- dags.gitSync.sshKnownHosts is set + no other values in the values file

It does **not** work when
- dags.gitSync.sshKnownHosts is set + my additional values configured

**Comment by samudurand** [comment]
Check above, I just arrived to the exact same conclusion xD

**Comment by samudurand** [comment]
The problem is that I really need that setting...

**Comment by bensta** [comment]
Ha, thanks. Sorry, did not see your comment before. 

My values file is quite extensive, so to figure out which **other** config (together with the known hosts setting) messes it up is quite laborious. 

Do you have a minimal yaml file that shows the error?

**Comment by samudurand** [comment]
If you use `helm3 template` with any helm chart based on this airflow chart, I think this should be more than enough to demonstrate:

```
airflow:
  dags:
    gitSync:
      enabled: true
      sshKnownHosts: |-
        github.com ssh-rsa xxxxxxxx
```

**Comment by bensta** [comment]
Weird. For me this works: (with helm install)

```yaml
dags:
  gitSync:
    sshKnownHosts: |+
      |1|M2SKp/Njffm11qkN1ANeQHTRy6c=|xamHkDNIpukUcGyDdObIK04kWYE= ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAubiN81eDcafrgMeLzaFPsw2kNvEcqTKl/VqLat/MaB33pZy0y3rJZtnqwR2qOOvbwKZYKiEO1O6VqNEBxKvJJelCq0dTXWT5pbO2gDXC6h6QDXCaHo6pOHGPUy+YBaGQRGuSusMEASYiWunYN0vCAI8QaXnWMXNMdFP3jHAJH0eDsoiGnLPBlBp4TNm6rYI74nMzgz3B9IikW4WVK+dc8KZJZWYjAuORU3jc1c/NPskD2ASinf8v3xnfXeukU0sJ5N6m5E8VLjObPEO+mN2t/FZTMZLiFqPWc/ALSqnMnnhwrNi2rbfg/rd/IpL8Le3pSBne8+seeFVBoGqzHM9yXw==
```

**Comment by bensta** [comment]
> If you use `helm3 template` with any helm chart based on this airflow chart, I think this should be more than enough to demonstrate:
> 
> ```
> airflow:
>   dags:
>     gitSync:
>       enabled: true
>       sshKnownHosts: |-
>         github.com ssh-rsa xxxxxxxx
> ```

Btw, I think your yaml is wrong. *dags* is not a child of airflow but at the top level.

**Comment by samudurand** [comment]
You yaml is missing the `enabled: true`, if you don't use that I think the gitsync configuration is not used, I tried without it and the gitsync doesn't start, but the build works.

And for my yaml, i don't know exactly why but in our chart (depending on airflow helm), we have a top level `airflow` object, which makes the yaml look like that: 

```
airflow:
  airflow:
  dags:
  ...
```

**Comment by bensta** [comment]
you are right, thanks. I was able to reproduce the error with your minimal config.
I tried a workaround using extraEnv variables, but unfortunately, those are not set in the sidecar.

**Comment by samudurand** [comment]
Well I got a good news (for me at least), it appears the known hosts is not exactly necessary in my case. I hope this will be fixed but it's not urgent in our case.

**Comment by meseta** [comment]
just bumped into this issue as well. I'm not a helm expert, but it appears to be that wherever `include "airflow.volumes" (dict ...)` is used, the `(dict )` is overriding the context with a custom object that doesn't include `.Release`, which then causes the "nil pointer evaluating interface {}.Name" error when it tries to evaluate `.Release.name`

So for example, starting at `worker/worker-statefulset.yaml` where it includes the "airflow.volumes", passing a custom context object that includes "Values" **but does not include "Release"**
https://github.com/airflow-helm/charts/blob/8611c766992a9f7156aad450f17beb2213cf667c/charts/airflow/templates/worker/worker-statefulset.yaml#L170

This includes the below, which creates the volume `git-known-hosts` where ssh known hosts are stored", it includes "airflow.fullname" as part of the secretName

https://github.com/airflow-helm/charts/blob/8611c766992a9f7156aad450f17beb2213cf667c/charts/airflow/templates/_helpers/pods.tpl#L235-L240

Here, "airflow.fullname" is defined as using `.Release.Name` but `.Release` was never passed to it as the context over in  `worker/worker-statefulset.yaml`

https://github.com/airflow-helm/charts/blob/8611c766992a9f7156aad450f17beb2213cf667c/charts/airflow/templates/_helpers/common.tpl#L5-L7

One solution is to pass "Release" .Release inside that dict. For example:
```
{{- $volumes := include "airflow.volumes" (dict "Release" .Release "Values" .Values "extraPipPackages" $extraPipPackages "extraVolumes" $extraVolumes) }}
```

However, this would need to be updated in all four places this is used (flower-deployment.yaml, scheduler-deployment.yaml, webserver-deployment.yaml, worker-statefulset.yaml, and one more in a doccumentation comment in _helpers/pod.tpl), I'm not sure if there's a better way for "allow.fullname" to access the root context `.Release.Name`. Does any helm experts here have a better way to do this? or shall we go ahead and PR changes to add "Release" to the include context in all four places?

**Comment by rolanddb** [comment]
I can confirm that the issue appeared for me when I included this snippet from the readme:
```
dags:
  persistence:
    enabled: true
    storageClass: default
    accessMode: ReadOnlyMany
    size: 1Gi
```

**Comment by meseta** [comment]
@rolanddb Fortunately, my PR above should account for this as well, as the issue should be similar. Both persistence and git-sync create volume mounts here, where persistence uses `include "airflow.fullname"` in the `claimName` (line 213) while git-sync uses it in the secretName (line 238)

https://github.com/airflow-helm/charts/blob/8611c766992a9f7156aad450f17beb2213cf667c/charts/airflow/templates/_helpers/pods.tpl#L205-L241

Let's hope the PR can be merged (or some equivalent fix)

**Comment by thesuperzapper** [comment]
@samudurand can you confirm if this issue is resolved after version `8.0.4`?
